devito/dse/graph.py

```diff
@@ -87,148 +101,190 @@ def construct(self, rule):
         """
         reads = set(self.reads) - set(rule.keys()) | set(rule.values())
         rhs = self.rhs.xreplace(rule)
-        return Temporary(self.lhs, rhs, reads=reads, readby=self.readby,
-                         scope=self.scope)
+        return Temporary(self.lhs, rhs, reads=reads, readby=self.readby)
 
     def __repr__(self):
-        return "DSE(%s, reads=%s, readby=%s)" % (super(Temporary, self).__repr__(),
-                                                 str(self.reads), str(self.readby))
+        reads = '[%s%s]' % (', '.join([str(i) for i in self.reads][:2]), '%s')
+        reads = reads % ('' if len(self.reads) <= 2 else ', ...')
+        readby = '[%s%s]' % (', '.join([str(i) for i in self.readby][:2]), '%s')
+        readby = readby % ('' if len(self.readby) <= 2 else ', ...')
+        return "Temp(key=%s, reads=%s, readby=%s)" % (self.lhs, reads, readby)
 
 
 class TemporariesGraph(OrderedDict):
 
     """
-    A temporaries graph built on top of an OrderedDict.
+    A temporaries graph represents an ordered sequence of operations.
+
+    The operations may involve scalars and indexed objects (arrays). The indices
+    of the indexed objects represent either "space" or "time" dimensions.
```

Comment:
this will be generalised; in essence, space dimensions are all parallel dimensions; time dimensions are the sequential ones; this is often used as compiler jargon